version: "3.7"

volumes:
  postgres-data:
    driver: local

services:
  db:
    container_name: litefarm-db
    image: postgres:13
    ports:
      - "5433:5432"
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
    healthcheck:
      test: [ "CMD", "pg_isready", "-U", "postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
      - ~/docker/volumes/postgres:/var/lib/postgresql/data
  backend:
    container_name: litefarm-api
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3
    build:
      context: ./packages/api
      dockerfile: prod.Dockerfile
    volumes:
      - /etc/letsencrypt:/usr/src/letsencrypt
    ports:
      - "5000:5000"
    environment:
      WAIT_HOSTS: litefarm-db:5432
      NODE_ENV: integration
      DEV_DATABASE_HOST: ${DEV_DATABASE_HOST}
      DEV_DATABASE: ${DEV_DATABASE}
      DEV_DATABASE_USER: ${DEV_DATABASE_USER}
      DEV_DATABASE_PASSWORD: ${DEV_DATABASE_PASSWORD}
  frontend:
    container_name: litefarm-web
    build:
      context: ./packages/webapp
      dockerfile: prod.Dockerfile
    volumes:
      - /etc/letsencrypt:/etc/letsencrypt
    ports:
      - "80:80"
      - "443:443"
